### ISSUE TITLE
Balloon radio buttons should be replaced with an ABSwitch

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/balloons-and-static-electricity#112]
For accessibility, it makes more sense for the two balloon radio buttons to be represented as a toggle button.  This issue continues https://github.com/phetsims/balloons-and-static-electricity/issues/109.

To accomplish this, it would be best to replace the balloon radio buttons with an ABSwitch, since ABSwitches would be represented by an accessible toggle button.  Elements of the screen view should match those of the Parallel DOM.  These radio buttons act as an ABSwitch now anyway.

This should be straightforward replacement so I am going to do it in the accessible-instance branch.  However, I think this would be a beneficial interface change for the simulation so I recommend that it be added as part of the redeploy.

**Comment by jessegreenberg** [comment]
Here is what it looks like now:
![abswitch](https://cloud.githubusercontent.com/assets/6396244/12278815/71916e08-b940-11e5-87a0-4ac8ac18662b.PNG)

It takes up a bit more real estate, but looks good overall.

**Comment by jessegreenberg** [comment]
The accessible content for the toggle button should look something like this:

```html
<div id=toggle-button-container>
   <input type="button" value="Add balloon" aria-pressed="false" aria-describedby="balloons-toggle-description"  id="toggle-button">
   <p>Translatable description of toggle button function...</p>
</div>

This is a structure that we have been wanting for multiple input elements, but has been tricky because it require structuring multiple elements under the single scenery node.  Also, the domElement that is passed into ```initializeAccessiblePeer``` must correspond to a scenery node for the node to receive the focus highlight.

I started to add accessible content to sun/ABSwitch, but realized that the above structure could be created if we create a new node under the parent, which just handles accessibility.  I am imagining something like:

ABSwitch ---> Represented by <div>
   accessibleABSwitchNode --> scenery node Child of ABSwitch, represented by <input>
   accessibleDescriptionNode --> scenery node child of ABSwitch, represented by <p>

The accessible content of ABSwitch can correctly set up the aria-describedby attributes with unique id's for the ABSwitch node.

I am going to play with this locally in a branch because I am aware that adding hierarchy could impact performance.

**Comment by jessegreenberg** [comment]
Commit above replaces for the ABSwitch, and creates an AccesibleABSwitchNode, to structure the DOM as described above.  The element looks like this in the parallel DOM:

```html
<div id="toggle-container14-35-417-492-484-473-468">
   <input type="button" id="abswitch-14-35-417-492-484-473-468-472" aria-describedby="button-description-468" value="‪Remove green balloon‬" aria-pressed="true">
   <p id="button-description-468">‪Toggle to conduct experiments with two balloons or just one.‬</p></div>
```

**Comment by jessegreenberg** [comment]
Works great with NVDA + Firefox and Chrome, and IE.  Testing JAWS next.

**Comment by jessegreenberg** [comment]
Works great with JAWS + Firefox and Chrome.  I experience lots of issues with JAWS 17 + IE 11, I will need to investigate that in a separate issue.

**Comment by jessegreenberg** [comment]
Things seem to be working for accessibility, but I realize that the balloons colors were optimized so that they were most visible over the blue background.  Will need to verify and make changes to this design suggestion before merging with master.

**Comment by terracoda** [comment]
@jessegreenberg looks perfectly reasonable. In your _Common UI Elements_ doc, I used a `button` in the sample code, but `input type="button"` makes good sense. 
Now that it's a switch, I think I the labels need to be "Single Balloon Experiment" and "Two Balloon Experiment", no rush. I want to listen to it in more environments first.

**Comment by terracoda** [comment]
@jessegreenberg, I'm just looking a little more closely at the code for the new Balloon A-B Switch. It doesn't look like there is label.

```<div id="toggle-container14-35-417-492-484-473-468">
   <input type="button" id="abswitch-14-35-417-492-484-473-468-472" aria-describedby="button-description-468" value="‪Remove green balloon‬" aria-pressed="true">
   <p id="button-description-468">‪Toggle to conduct experiments with two balloons or just one.‬</p></div>```

It is better to use a labeling technique rather than aria-describedby. Labels are for essential content and descriptions are important, but not essential. Because there is no overt text on the switch, the `aria-label` attribute is the next best thing to a real `label` element. An aria-label attribute provides an "accessible name".
Here's the change I am suggesting, note that I change the content of the label. Value for the B part of the switch would be "Two Balloon Experiment."
```<div id="toggle-container14-35-417-492-484-473-468">
   <input type="button" id="abswitch-14-35-417-492-484-473-468-472" aria-label="Single Balloon Experiment" value="‪Single Balloon Experiment‬" aria-pressed="true">
   <p id="button-description-468">‪Toggle to conduct experiments with two balloons or just one.‬</p></div>```

**Comment by jessegreenberg** [comment]
Excellent suggestion @terracoda, I will change the HTML of the PDOM to add an aria-label with the suggested strings.

**Comment by jessegreenberg** [comment]
Made the change to the parallel DOM, the code now looks like this.

```html
<div id="toggle-container14-35-417-492-484-473-468">
<input type="button" id="abswitch-14-35-417-492-484-473-468-472" value="‪Remove green balloon‬" aria-label="‪Remove green balloon‬" aria-pressed="true">
</div>
```

**Comment by jessegreenberg** [comment]
Now that there is no accessible description, it seems like the containing div can be removed.  This is beneficial, we no longer need a containing scenery node to structure the parallel DOM.

**Comment by jessegreenberg** [comment]
Now I am confused about the state of the aria-label.  Now that the aria-label updates with the button state, it seems confusing because both the label and the toggle state are changing at the same time.

It seems like the aria-label should remain constant.  Otherwise, the screen reader would read something like 'Single Balloon Experiment, Pressed', when there are two balloons on screen.

@terracoda, is this understanding correct?

**Comment by jessegreenberg** [comment]
After discussion at a 1/22/16 meeting:

The aria-label should not change with the toggle state.  When two balloons are on the screen, the description should be like 'Two balloon experiment, pressed'

**Comment by terracoda** [comment]
Thanks @jessegreenberg for clarifying the way the "pressed" states on a toggle are read out!

**Comment by jessegreenberg** [comment]
Labels for the AccessibleABSwitchNode are now static, the aria-pressed state is the only variable.

**Comment by jessegreenberg** [comment]
This is done, but leaving open to discuss if this design change can propagate to master.

**Comment by jessegreenberg** [comment]
We reverted to the visual look of the 'scene selection' style, but kept the switch representation in the PDOM in #112. Closing.
